pico/services/src/main/java/io/helidon/pico/services/DefaultPicoServices.java

```diff
@@ -340,7 +332,7 @@ public synchronized boolean reset(boolean deep) {
                 state.reset(true);
                 initializingServicesStarted.set(false);
                 initializingServicesFinished.set(false);
-                initializationCallingContext = null;
+                initializationCallingContext = Optional.empty();
```

Comment:
Why is this field even an optional? 
Either use a non-optional & nulls, or use Optional, but then utilize the Optional methods (such as on line 370, where instead of a ternary expression, you could use functional style to do the same

Reply:
I'll switch to using null in the next commit.